### DISCUSSION TITLE
Rebase on newer Mutter

### DISCUSSION THREAD

**Discussion by axel358** [linuxmint/discussions#251] [Development]
I think that a lot of the bugs related to Wayland could be solved by updating Muffin. In my particular case fractional scaling is broken on both x11 and Wayland, screen tearing and glitches respectively. I can't reproduce these bugs on gnome 45. Reading through the muffin n Wayland issues list there are other bugs related to missing protocols that have been merged on upstream mutter already, like Waydroid color fix.

**Comment by kacperpaczos** [comment]
Maybe you should try to do it?

**Comment by Fantu** [comment]
A larger rebase was done not long ago but unfortunately on an "already old" version when it started, based on mutter of the previous ubuntu lts (20.04), and completed with the release of muffin 5.4. To avoid something similar, done I suppose for compatibility with the versions of Mint supported at the time, I think it might be better to wait in case it is not possible to do a rebase on the next version currently in development (and which will be used in Ubuntu 24.04). 
In any case, before starting it is best to ask that no one else is already working on it and ask @clefebvre @mtwebster if he would be okay with using a possible rebase if it were done or even if there were some relevant changes (involving other cinnamon components) to discuss before with them.

**Comment by mtwebster** [comment]
We didn't rebase onto anything newer at the time because doing so would have required us to break every applet, desklet and extension. There's no point in doing so currently, either (there's plenty else holding us back with wayland).

**Comment by axel358** [comment]
I wouldn't say that "there's no point on doing so", i can't use fractional scaling on my new hardware, and i can't use my new hardware without fractional scaling. I've temporarily moved to GNOME but i don't think that's the solution either. I didn't create an issue specifically about scaling because i thought this approach could potentially fix other issues

**Reply by rcalixte** [reply]
Are you referring to the Wayland or X11 session that is unable to use fractional scaling? The X11 session is probably the main target to be addressed. Could you create an Issue with the details, including your hardware configuration?

**Reply by axel358** [reply]
On x11 there's screen screen tearing that doesn't seem to go away with the usual fix for Intel. On Wayland there are several glitches, flickering and other visual artifacts that prevent the session from being usable either. I'll create an issue for x11 with the details then
